jwst/extract_1d/extract1d.py

```diff
@@ -85,6 +90,14 @@ def extract1d(image, lambdas, disp_range,
         For each column, this is the number of pixels that were added
         together to get `countrate`.
     """
+    if not subtract_background:
```

Comment:
This function can be called in a loop over planes of a 3-D array, e.g. TSO data, which could result in a very large number of info messages.  For the case that the reference is a JSON file, I would suggest checking the `subtract_background` flag in extract.py, in the `__init__` for `ExtractModel`, in this section:

    # Coefficients for source (i.e. target) and background limits and
    # corresponding polynomial functions.
    self.src_coeff = copy.deepcopy(src_coeff)
    self.bkg_coeff = copy.deepcopy(bkg_coeff)

If `bkg_coeff` was specified in the reference file, but the user said to not do background subtraction, just set `self.bkg_coeff` to `None` instead of copying `bkg_coeff`.  Or if the user said that background subtraction should be done but `bkg_coeff` was not defined in the reference file, you could log a warning to say that it can't be done because there's no information as to where to extract the background (as you did here).